packages/components/src/button/index.native.js

```diff
@@ -91,108 +52,130 @@ export function Button( props ) {
 		label,
 		shortcut,
 		tooltipPosition,
+		styles = {},
 	} = props;
 
-	const isDisabled = ariaDisabled || disabled;
+	const colorScheme = usePreferredColorScheme();
 
-	const buttonViewStyle = {
-		opacity: isDisabled ? 0.3 : 1,
-		...( fixedRatio && styles.fixedRatio ),
-		...( isPressed ? styles.buttonActive : styles.buttonInactive ),
-	};
+	const isDisabled = ariaDisabled || disabled;
 
-	const states = [];
-	if ( isPressed ) {
-		states.push( 'selected' );
-	}
+	const themedStyle = usePreferredColorSchemeStyleBem( {
+		...baseStyles,
+		...styles,
+	} );
 
-	if ( isDisabled ) {
-		states.push( 'disabled' );
-	}
+	/* eslint-disable dot-notation */
+	const containerStyle = themedStyle[ 'button__container' ];
+
+	const buttonStyle = useMemo(
+		() => ( {
+			opacity: isDisabled ? 0.3 : 1,
+			...( fixedRatio && { aspectRatio: 1 } ),
+			...( isPressed
+				? themedStyle[ 'button__button--active' ]
+				: themedStyle[ 'button__button--inactive' ] ),
+		} ),
+		[ fixedRatio, isDisabled, isPressed ]
+	);
 
-	const subscriptInactive = getStylesFromColorScheme(
-		styles.subscriptInactive,
-		styles.subscriptInactiveDark
+	const subscriptStyle = useMemo(
+		() =>
+			isPressed
+				? themedStyle[ 'button__subscript--active' ]
+				: themedStyle[ 'button__subscript--inactive' ],
+		[ isPressed ]
 	);
 
+	const iconColor = themedStyle[ 'button__icon' ]?.color;
+	/* eslint-enable dot-notation */
+
 	const newChildren = Children.map( children, ( child ) => {
 		return child
 			? cloneElement( child, {
-					colorScheme: props.preferredColorScheme,
+					colorScheme,
 					isPressed,
 			  } )
 			: child;
 	} );
 
-	// Should show the tooltip if...
-	const shouldShowTooltip =
-		! isDisabled &&
-		// an explicit tooltip is passed or...
-		( ( showTooltip && label ) ||
-			// there's a shortcut or...
-			shortcut ||
-			// there's a label and...
-			( !! label &&
-				// the children are empty and...
-				( ! children ||
-					( isArray( children ) && ! children.length ) ) &&
-				// the tooltip is not explicitly disabled.
-				false !== showTooltip ) );
-
-	const newIcon = icon
-		? cloneElement( <Icon icon={ icon } size={ iconSize } />, {
```

Comment:
I'm not sure why `cloneElement` was used here.  From my understanding the main use is only to update the props of a child component. Since we are creating the `Icon` there should be no need to clone it.